ui/src/pages/QuickSetup/AddPartiesPage.tsx

```diff
@@ -260,40 +273,51 @@ const AdminLedger = (props: { adminCredentials: Credentials; onComplete: () => v
     const offerRegulatorServices = async () => {
       await Promise.all(
         userParties.map(async party => {
-          if (
-            !regulatorServices.find(c => c.payload.customer === party.party) &&
-            !regulatorServiceOffers.find(c => c.payload.customer === party.party)
-          ) {
-            return await offerRegulatorService(party.party);
+          if (!regulatorServiceOffers.find(c => c.payload.customer === party.party)) {
+            await offerRegulatorService(party.party);
           }
         })
       );
     };
 
-    if (
-      operatorServiceLoading ||
-      regulatorRolesLoading ||
-      regulatorServicesLoading ||
-      regulatorServiceOffersLoading
-    ) {
+    if (operatorServiceLoading || regulatorRolesLoading || regulatorServiceOffersLoading) {
       return;
     }
 
-    if (
-      userParties.every(
-        p =>
-          !!regulatorServiceOffers.find(c => c.payload.customer === p.party) ||
-          !!regulatorServices.find(contract => contract.payload.customer === p.party)
-      )
-    ) {
-      return onComplete();
+    async function deployAllTriggers() {
+      if (isHubDeployment && parties.length > 0) {
+        const artifactHash = TRIGGER_HASH;
+
+        if (!artifactHash || !adminCredentials) {
+          return;
+        }
+
+        Promise.all(
+          [
+            ...parties.filter(p => p.party !== publicParty),
+            {
+              ...adminCredentials,
+            },
+          ].map(p => {
+            return deployAutomation(
+              artifactHash,
+              MarketplaceTrigger.AutoApproveTrigger,
+              p.token,
+              publicParty
+            );
+          })
+        );
+      }
     }
+
     if (operatorService.length === 0) {
       createOperatorService();
     } else if (regulatorRoles.length === 0) {
       createRegulatorRole();
     } else {
       offerRegulatorServices();
+      deployAllTriggers();
+      return onComplete();
```

Comment:
I think it is better to deploy triggers here, so that there is no race condition between creating and accepting regulator offers - and the UI can reliably check if all parties have been offered the role

Reply:
Also, here the party list is fully loaded where as in quick setup that was not a guarantee and sometimes required a refresh